src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -166,18 +173,36 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
             try
             {
-                _logger.LogTrace("Beginning receive loop");
+                _logger.LogTrace("Beginning receive loop.");
 
                 while (await Input.WaitToReadAsync())
                 {
+                    if (_connectionState != ConnectionState.Connected)
```

Comment:
How correct does this need to be? Doesn't it need to be synchronized so that we never call a user callback if they are disconnected?